### PR TITLE
Add logging to empty catch blocks to fix PMD violations, issue #744

### PR DESCRIPTION
**Pull request by mkordas**
#### Use Guava Closeables to manage closing exceptions 

>While it's not safe in the general case to ignore exceptions that are thrown when closing an I/O resource, it should generally be safe in the case of a resource that's being used only for reading, such as a Reader. Unlike with writable resources, there's no chance that a failure that occurs when closing the reader indicates a meaningful problem such as a failure to flush all bytes to the underlying resource.

`Reader` and `InputStream` instances are closed using `Closeables.closeQuietly()`, while `RandomAccessFile` is closed with `Closeables.close()` that throws `IOException` and needs to be handled.

From Javadoc:
>```java
public static void close(@Nullable
         Closeable closeable,
         boolean swallowIOException)
                  throws IOException
```
>Closes a Closeable, with control over whether an IOException may be thrown. This is primarily useful in a finally block, where a thrown exception needs to be logged but not propagated (otherwise the original exception will be lost).
If swallowIOException is true then we never throw IOException but merely log it.

>Example:
```java
   public void useStreamNicely() throws IOException {
     SomeStream stream = new SomeStream("foo");
     boolean threw = true;
     try {
       // ... code which does something with the stream ...
       threw = false;
     } finally {
       // If an exception occurs, rethrow it only if threw==false:
       Closeables.close(stream, threw);
     }
   }
```

Moreover, `Closeables.close()` and  `Flushables.flush()` are used to flush and close `OutputStream`.

#### Add logging to empty catch blocks to fix PMD violations, issue #744 

All violations of PMD rule [EmptyCatchBlock](http://pmd.sourceforge.net/pmd-5.2.3/pmd-java/rules/java/empty.html#EmptyCatchBlock) are fixed by logging exceptions.

#### Add unit tests for class base name util to improve coverage

Just to bump up coverage in `Utils` that was reduced by removing code.

#### General information

After this pull request will be merged, PMD report on Checkstyle's site won't generate anymore, as all violations are fixed. Then I'm going to bind PMD to `verify` or earlier phase to get fast feedback.

### CODE REVIEW

**Comment by romani** [comment]
please:
1) use http://docs.guava-libraries.googlecode.com/git-history/v18.0/javadoc/src-html/com/google/common/io/Closeables.html#line.71

2) please add some UTs to atleast not reduce a coverage, we should not go down, even changes are minimal.

**Comment by mkordas** [comment]
1. Where should I use Guava's `Closeables.close`? Please note that this method has declared `throws IOException` - it's different from our `closeQuietly`. It requires additional error handling which is hard to test without mocking framework.
2. OK, I'll see what can be done.

**Comment by romani** [comment]
> Where should I use Guava's Closeables.close? 

Ok ,sorry, I see a difference now. 
That is even more sigh of bad design or bad usage. 
Util methods should not have Logging abiities, logging is part of business logic, so " private static final Log LOG" is not good approach . We need to deside. If we ignore exception - that method is definitely Util method. If we want to log smth, we need to use custom code in all places where closeQuietly is user right now. I do not even see a reason why we should swollow that exception - we need to print to log as much details on failure as possible.

File: src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java [hunk 1]
```diff
@@ -91,7 +97,7 @@ private Object doMakeObject(String name)
                 return createObject(className);
             }
             catch (final CheckstyleException ex) {
-                // keep looking
+                LOG.debug("Keep looking, ignoring exception", ex);
```
Comment (damianszczepanik) [inline src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java]: By merging those two blocks you can increase code coverage and (based on cobertura output) cover this fully.

Comment (mkordas) [inline src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java]: Thanks Damian, I'll try that.

**Comment by romani** [comment]
@ mkordas , lets finish that PR, and start preparation for release 6.5.

File: src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java (hunk 1 above)
Comment (mkordas) [inline src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java]: Hmm... in this case I don't know what should I merge.

Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java]: no need merge anything, that block are not related even they looks close on diff

**Comment by mkordas** [comment]
@romani, I've reworked the PR, please review.

**Comment by mkordas** [comment]
Coverage in `Utils` can be increased further by deleting deprecated `com.puppycrawl.tools.checkstyle.Utils#getLines(java.lang.String)`. This class is not in `api` anymore anyway. Should I submit an issue and PR for that?

**Comment by romani** [comment]
yes, please do that in other issue, no need to put all refactoring in one issue.
All deprecated method should be removed eventually.
